src/app/components/counter/counter.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, forwardRef, Input, OnInit } from '@angular/core';
+import { ControlValueAccessor, FormBuilder, NG_VALUE_ACCESSOR, Validators } from '@angular/forms';
+
+import { counterStyle } from 'src/app/core/models/counter.enum';
+
+@Component({
+  selector: 'app-counter',
+  templateUrl: './counter.component.html',
+  styleUrls: ['./counter.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CounterComponent),
+      multi: true
+    }
+  ]
+})
+export class CounterComponent implements ControlValueAccessor, OnInit {
+
+  counterStyles = counterStyle;
+
+  @Input() counterStyle?: number;
+  @Input() incrementNumber: number = 1;
+  disabled: boolean = false;
+
+  counterValue: number = 0;
+
+  onChange = (value: any) => { };
+  onTouche = () => { };
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.counterForm.valueChanges.subscribe((value) => {
```

Comment:
we need to unsubscribe in destroy

use take until strategy:  https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe